src/components/com_tjcertificate/site/controllers/externalcertificate.php

```diff
@@ -0,0 +1,264 @@
+	 */
+	public function publish()
+	{
+		parent::publish();
+
+		$this->setRedirect('index.php?option=com_tjcertificate&view=certificates&layout=my');
+	}
+
+	/**
+	 * Method to download certificate.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function downloadCertificate()
+	{
+		// CSRF token check
+		Session::checkToken('get') or jexit(Text::_('JINVALID_TOKEN'));
+
+		$user = Factory::getUser();
+		$app  = Factory::getApplication();
+		$params = ComponentHelper::getParams('com_tjcertificate');
+
+		// Validate user login.
+		if (empty($user->id))
+		{
+			$return = base64_encode((string) Uri::getInstance());
+			$login_url_with_return = Route::_('index.php?option=com_users&return=' . $return);
+			$app->enqueueMessage(Text::_('JERROR_ALERTNOAUTHOR'), 'notice');
+			$app->redirect($login_url_with_return, 403);
+		}
+
+		$mediaId  = $app->input->get('mediaId', '', 'INT');
+
+		$config              = array();
+		$config['mediaId']   = $mediaId;
+
+		// Assign client id as Campaign Id or Report Id or Giveback Id
+		$config['client_id'] = $app->input->get('certificateId', '', 'INT');
+		$config['client']    = 'com_timelog.activity';
+		$mediaAttachmentData = TJMediaXref::getInstance($config);
+
+		$folderName          = explode('.', $mediaAttachmentData->media->type);
+		$downloadPath        = JPATH_SITE . '/' . $params->get('file_path', 'media/com_tjcertificate/external');
+
+		// Making File Download path For e.g /file mime type + 's'/text.pdf Here mime type like application + s this is folder name
+		$downloadPath        = $downloadPath . '/' . $folderName[0] . '/' . $mediaAttachmentData->media->source;
+
+		$media = TJMediaStorageLocal::getInstance();
+		$media->downloadMedia($downloadPath);
+	}
+
+	/**
+	 * Function to delete the timelog activity attachment
+	 *
+	 * @return  void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function deleteAttachment()
+	{
+		// Prevent CSRF attack
+		Session::checkToken('get') or jexit(Text::_('JINVALID_TOKEN'));
+
+		// Get the current user id
+		$user = Factory::getuser();
+		$app  = Factory::getApplication();
+
+		if (!$user->id)
+		{
+			return false;
+		}
+
+		$params   = ComponentHelper::getParams('com_tjcertificate');
+		$filePath = $params->get('file_path', 'media/com_tjcertificate/external');
+		$clientId = $app->input->get('activityId', '', 'INT');
+		$mediaId  = $app->input->get('mediaId', '', 'INT');
+
+		if (!$mediaId && !$clientId)
+		{
+			throw new Exception(Text::_('JERROR_ALERTNOAUTHOR'), 403);
+		}
+
+		$model  = $this->getModel();
+		$return = $model->deleteMedia($mediaId, $filePath, 'com_tjcertificate', $clientId);
+
+		$result = array();
+		$result['success'] = true;
+		$result['message'] = Text::_('COM_TJCERTIFICATE_ATTACHMENT_DELETED_SUCCESSFULLY');
+
+		if ($return == false)
+		{
+			$result['success'] = false;
+			$result['message'] = Text::_('COM_TJCERTIFICATE_ATTACHMENT_DELETED_FAILED');
+		}
+
+		echo json_encode($result);
```

Comment:
move to json type controller i.e. externalcertificate.json.php

Reply:
done